controllers/machinehealthcheck_controller.go

```diff
@@ -0,0 +1,194 @@
+		WithOptions(options).
+		Build(r)
+
+	if err != nil {
+		return errors.Wrap(err, "failed setting up with a controller manager")
+	}
+
+	// Add index to MachineHealthCheck for listing by Cluster Name
+	if err := mgr.GetCache().IndexField(&clusterv1.MachineHealthCheck{},
+		mhcClusterNameIndex,
+		r.indexMachineHealthCheckByClusterName,
+	); err != nil {
+		return errors.Wrap(err, "error setting index fields")
+	}
+
+	r.controller = controller
+	r.recorder = mgr.GetEventRecorderFor("machinehealthcheck-controller")
+	return nil
+}
+
+func (r *MachineHealthCheckReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error) {
+	ctx := context.Background()
+	logger := r.Log.WithValues("machinehealthcheck", req.Name, "namespace", req.Namespace)
+
+	// Fetch the MachineHealthCheck instance
+	m := &clusterv1.MachineHealthCheck{}
+	if err := r.Client.Get(ctx, req.NamespacedName, m); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+
+		// Error reading the object - requeue the request.
+		logger.Error(err, "Failed to fetch MachineHealthCheck")
+		return ctrl.Result{}, err
+	}
+
+	cluster, err := util.GetClusterByName(ctx, r.Client, m.Namespace, m.Spec.ClusterName)
+	if err != nil {
+		logger.Error(err, "Failed to fetch Cluster for MachineHealthCheck")
+		return ctrl.Result{}, errors.Wrapf(err, "failed to get Cluster %q for MachineHealthCheck %q in namespace %q",
+			m.Spec.ClusterName, m.Name, m.Namespace)
+	}
+
+	// Return early if the object or Cluster is paused.
+	if util.IsPaused(cluster, m) {
+		logger.V(3).Info("reconciliation is paused for this object")
+		return ctrl.Result{}, nil
+	}
+
+	// Initialize the patch helper
+	patchHelper, err := patch.NewHelper(m, r.Client)
+	if err != nil {
+		logger.Error(err, "Failed to build patch helper")
+		return ctrl.Result{}, err
+	}
+
+	defer func() {
+		// Always attempt to patch the object and status after each reconciliation.
+		if err := patchHelper.Patch(ctx, m); err != nil {
+			reterr = kerrors.NewAggregate([]error{reterr, err})
+		}
+	}()
+
+	// Reconcile labels.
+	if m.Labels == nil {
+		m.Labels = make(map[string]string)
+	}
+	m.Labels[clusterv1.ClusterLabelName] = m.Spec.ClusterName
+
+	result, err := r.reconcile(ctx, cluster, m)
+	if err != nil {
+		logger.Error(err, "Failed to reconcile MachineHealthCheck")
+		r.recorder.Eventf(m, corev1.EventTypeWarning, "ReconcileError", "%v", err)
+
+		//TODO(JoelSpeed): Determine how/when to requeue requests if errors occur within r.reconcile
+	}
+
+	return result, nil
+}
+
+func (r *MachineHealthCheckReconciler) reconcile(_ context.Context, cluster *clusterv1.Cluster, m *clusterv1.MachineHealthCheck) (ctrl.Result, error) {
+	// Ensure the MachineHealthCheck is owned by the Cluster it belongs to
+	m.OwnerReferences = util.EnsureOwnerRef(m.OwnerReferences, metav1.OwnerReference{
```

Comment:
@detiber controllerutil.SetControllerReference here?

Reply:
I think it's probably safer to leave these types of ownerReferences as non-controller references for now.